src/venue/venue.service.ts

```diff
@@ -86,36 +109,79 @@ export class VenueService {
   }
 
   async update(venueId: number, updateVenueData: UpdateVenueDto) {
-    const { amenitiesIds, ...venueData } = updateVenueData;
-    try {
-      const updatedVenue = await this.prismaService.venue.update({
-        where: { id: venueId },
-        data: venueData,
+    const { amenitiesIds, ...partialData } = updateVenueData;
+
+    const existingVenue = await this.prismaService.venue.findUnique({
+      where: { id: venueId },
+    });
+
+    if (!existingVenue) {
+      throw new NotFoundException(`Venue with ID ${venueId} not found`);
+    }
+
+    const streetNumber =
+      updateVenueData.streetNumber ?? existingVenue.streetNumber;
+    const streetName = updateVenueData.streetName ?? existingVenue.streetName;
+    const postalCode = updateVenueData.postalCode ?? existingVenue.postalCode;
+    const city = updateVenueData.city ?? existingVenue.city;
+
+    const shouldRecalculateLocation =
+      updateVenueData.streetNumber !== undefined ||
+      updateVenueData.streetName !== undefined ||
+      updateVenueData.postalCode !== undefined ||
+      updateVenueData.city !== undefined;
+
+    let latitude: number | undefined;
+    let longitude: number | undefined;
+
+    if (shouldRecalculateLocation) {
```

Comment:
It looks like you are about to implement very complex updating logic in a single method. Usually this kind of solutions are ending up to complex to follow and at the end it needs to be refeactored. My suggestion is to implement a lot of handlers for very specific updates. If you see that location update is not straightforward update, then you should probably create an updateLocation method in the service and use it. You can even go a step further and implement a very specific controller handler for a very specific update.